codex/purchasing.nim

```diff
@@ -53,7 +53,7 @@ proc populate*(purchasing: Purchasing,
   if result.ask.proofProbability == 0.u256:
     result.ask.proofProbability = purchasing.proofProbability
   if result.expiry == 0.u256:
-    result.expiry = (purchasing.clock.now().u256 + purchasing.requestExpiryInterval)
+    result.expiry = purchasing.requestExpiryInterval
```

Comment:
```suggestion
    result.expiry = await purchasing.market.getRequestExpiry(request.id)
```
I don't think we really need that default interval anymore. Not sure what it was for to be honest.

Reply:
Just a comment: `request.expiry` should never really be `0` as we checking for that in the rest api. I think this maybe isn't necessary at all, or actually we should be raising an error.